services/grpc/handler.go

```diff
@@ -48,21 +49,63 @@ func (h *Handler) SendEvent(ctx context.Context, req *pb.SendEventRequest) (*pb.
 	metrics.Increment("batches_read_total", fmt.Sprintf("status=success,conn_group=%s", identifier.Group))
 	h.sendEventCounters(req.Events, identifier.Group)
 
+	responseChannel := make(chan *pb.SendEventResponse, 1)
 	h.C.Collect(ctx, &collection.CollectRequest{
 		ConnectionIdentifier: identifier,
 		TimeConsumed:         timeConsumed,
 		SendEventRequest:     req,
+		AckFunc:              h.Ack(responseChannel, req.ReqGuid, identifier.Group),
 	})
+	return <-responseChannel, nil
```

Comment:
is this expected to always send nil error?

Reply:
The response has error codes and error reasons, so nil is being sent every time. This is a little tricky because we have tried to make it consistent with WebSocket and Rest responses.